tfjs-core/src/ops/browser.ts

```diff
@@ -154,6 +154,138 @@ function fromPixels_(
   return tensor3d(values, outShape, 'int32');
 }
 
+/**
+ * Creates a `tf.Tensor` from an image in async way.
+ *
+ * ```js
+ * const image = new ImageData(1, 1);
+ * image.data[0] = 100;
+ * image.data[1] = 150;
+ * image.data[2] = 200;
+ * image.data[3] = 255;
+ *
+ * (await tf.browser.fromPixelsAsync(image)).print();
+ * ```
+ *
+ * @param pixels The input image to construct the tensor from. The
+ * supported image types are all 4-channel. You can also pass in an image
+ * object with following attributes:
+ * `{data: Uint8Array; width: number; height: number}`
+ * @param numChannels The number of channels of the output tensor. A
+ * numChannels value less than 4 allows you to ignore channels. Defaults to
+ * 3 (ignores alpha channel of input image).
+ *
+ * @doc {heading: 'Browser', namespace: 'browser', ignoreCI: true}
+ */
+export async function fromPixelsAsync(
+  pixels: PixelData|ImageData|HTMLImageElement|HTMLCanvasElement|
+  HTMLVideoElement,
+  numChannels = 3) {
+ // Sanity checks.
+ if (numChannels > 4) {
+  throw new Error(
+      'Cannot construct Tensor with more than 4 channels from pixels.');
+}
+if (pixels == null) {
+  throw new Error('pixels passed to tf.browser.fromPixels() can not be null');
+}
+let isPixelData = false;
+let isImageData = false;
+let isVideo = false;
+let isImage = false;
+let isCanvasLike = false;
+if ((pixels as PixelData).data instanceof Uint8Array) {
+  isPixelData = true;
+} else if (
+    typeof (ImageData) !== 'undefined' && pixels instanceof ImageData) {
+  isImageData = true;
+} else if (
+    typeof (HTMLVideoElement) !== 'undefined' &&
+    pixels instanceof HTMLVideoElement) {
+  isVideo = true;
+} else if (
+    typeof (HTMLImageElement) !== 'undefined' &&
+    pixels instanceof HTMLImageElement) {
+  isImage = true;
+  // tslint:disable-next-line: no-any
+} else if ((pixels as any).getContext != null) {
+  isCanvasLike = true;
+} else {
+  throw new Error(
+      'pixels passed to tf.browser.fromPixels() must be either an ' +
+      `HTMLVideoElement, HTMLImageElement, HTMLCanvasElement, ImageData ` +
+      `in browser, or OffscreenCanvas, ImageData in webworker` +
+      ` or {data: Uint32Array, width: number, height: number}, ` +
+      `but was ${(pixels as {}).constructor.name}`);
+}
+if (isVideo) {
+  const HAVE_CURRENT_DATA_READY_STATE = 2;
+  if (isVideo &&
+      (pixels as HTMLVideoElement).readyState <
+          HAVE_CURRENT_DATA_READY_STATE) {
+    throw new Error(
+        'The video element has not loaded data yet. Please wait for ' +
+        '`loadeddata` event on the <video> element.');
+  }
+}
+// If the current backend has 'FromPixels' registered, it has a more
+// efficient way of handling pixel uploads, so we call that.
+const kernel = getKernel(FromPixels, ENGINE.backendName);
```

Comment:
Given this structure can this method just call fromPixels_ directly rather using ENGINE.runKernel/getKernel? I think you would get a lot of code-reuse that way and reduce duplication.

It would basically test if createImageBitmap is available (since its not supported in all browsers) do the `await imageBitmap = await createImageBitmap(pixels);` you have below and then forward to fromPixels. If createImageBitmap is not available it can return await fromPixels(...) that way it always has a fallback across browsers.

Reply:
@tafsiri fyi, #4374 is another patch to support `ImageBitmap` as a valid input of `fromPixels`

Reply:
@tafsiri Thanks for your reviewing and valuable suggestions! Totally agree with the direction to add this async op. And as @qjia7  said ,#4374 will be the base for this patch. So please take a look in that patch if you're available :)